server/routes/__tests__/check-your-details.spec.js

```diff
@@ -0,0 +1,129 @@
+const { getServer } = require('../../../.jest/setup')
+const {
+  submitGetRequest,
+  submitPostRequest
+} = require('../../__test-helpers__/server')
+const mockPolygons = require('../../services/__data__/mockPolygons.json')
+
+jest.mock('../../services/agol/getFloodZones')
+jest.mock('../../services/riskAdmin/isRiskAdminArea')
+jest.mock('../../services/agol/getContacts')
+jest.mock('../../services/address')
+jest.mock('@hapi/wreck')
+const wreck = require('@hapi/wreck')
+const user = {
+  fullName: 'John Smith',
+  email: 'john.smith@email.com'
+}
+
+const url = '/check-your-details'
+
+describe('Check your details page', () => {
+  beforeEach(() => {
+    wreck.post.mockResolvedValue({
+      payload: {
+        applicationReferenceNumber: '12345',
+        nextTask: 'SEND_CONFIRMATION_EMAIL'
+      }
+    })
+  })
+  describe('GET', () => {
+    const floodZoneGets = [
+      { polygon: mockPolygons.fz1_only, floodZone: '1' },
+      { polygon: mockPolygons.fz2_only, floodZone: '2' },
+      { polygon: mockPolygons.fz3_only, floodZone: '3' }
+    ]
+    floodZoneGets.forEach(({ polygon, floodZone }) => {
+      it(`Happy get request for a flood zone ${floodZone} information`, async () => {
+        const response = await submitGetRequest({ url: `${url}?polygon=${polygon}&fullName=${user.fullName}&recipientemail=${user.email}` }, 'Check your details before requesting your data')
+        document.body.innerHTML = response.payload
+        expect(document.querySelector('title').textContent).toContain('Check your details - Flood map for planning - GOV.UK')
+        expect(document.querySelector('.govuk-summary-list__row > dd.govuk-summary-list__value').textContent).toContain(user.fullName)
+        expect(document.querySelector('.govuk-summary-list__row:nth-child(2) > dd.govuk-summary-list__value').textContent).toContain(user.email)
+        expect(document.querySelector('.govuk-summary-list__row:nth-child(4) > dd.govuk-summary-list__value').textContent).toContain(floodZone)
+      })
```

Comment:
there is a helper I've added: 
 
const { assertCopy } = require('../../__test-helpers__/copy')
 
then 
        expect(document.querySelector('title').textContent).toContain('Check your details - Flood map for planning - GOV.UK')
could be replaced with: 
        assertCopy('title', 'Check your details - Flood map for planning - GOV.UK')
...etc